drivers/include/at.h

```diff
@@ -53,54 +54,85 @@ extern "C" {
 /**
  * @brief End of line character to send after the AT command.
  */
-#ifndef AT_SEND_EOL
-#define AT_SEND_EOL "\r"
+#if IS_ACTIVE(CONFIG_AT_SEND_EOL_WINDOWS)
+#define CONFIG_AT_SEND_EOL   "\r\n"
+#elif IS_ACTIVE(CONFIG_AT_SEND_EOL_UNIX)
+#define CONFIG_AT_SEND_EOL   "\n"
+#elif IS_ACTIVE(CONFIG_AT_SEND_EOL_MAC)
+#define CONFIG_AT_SEND_EOL   "\r"
+#endif
+
+#ifndef CONFIG_AT_SEND_EOL
+#define CONFIG_AT_SEND_EOL "\r"
+#endif
+
+/**
+ * @brief Enable this to disable check for echo after an AT
+ * command is sent.
+ */
+#ifdef DOXYGEN
+#define CONFIG_AT_SEND_SKIP_ECHO
 #endif
 
 /**
  * @brief Enable/disable the expected echo after an AT command is sent.
+ *
+ * @deprecated Use inverse @ref CONFIG_AT_SEND_SKIP_ECHO instead.
+ * Will be removed after 2021.01 release.
  */
 #ifndef AT_SEND_ECHO
+#if IS_ACTIVE(CONFIG_AT_SEND_SKIP_ECHO)
+#define AT_SEND_ECHO 0
+#else
 #define AT_SEND_ECHO 1
 #endif
+#endif
 
 /**
- * @brief 1st end of line character received (S3 aka CR character for a modem).
+ * @brief default OK reply of an AT device.
  */
-#ifndef AT_RECV_EOL_1
-#define AT_RECV_EOL_1   "\r"
+#ifndef CONFIG_AT_RECV_OK
+#define CONFIG_AT_RECV_OK "OK"
 #endif
 
 /**
- * @brief 1st end of line character received (S4 aka LF character for a modem).
+ * @brief default ERROR reply of an AT device.
  */
-#ifndef AT_RECV_EOL_2
-#define AT_RECV_EOL_2   "\n"
+#ifndef CONFIG_AT_RECV_ERROR
+#define CONFIG_AT_RECV_ERROR "ERROR"
 #endif
 
 /**
- * @brief default OK reply of an AT device.
+ * @brief Size of buffer used to process unsolicited result code data.
+ */
+#if defined(MODULE_AT_URC) || DOXYGEN
+#if defined(MODULE_AT_URC) || DOXYGEN
+#ifdef CONFIG_AT_BUF_SIZE_EXP
+#define CONFIG_AT_BUF_SIZE  (1<<CONFIG_AT_BUF_SIZE_EXP)
+#endif
```

Comment:
I think this could be like this
```suggestion
#ifdef CONFIG_AT_BUF_SIZE_EXP
#define CONFIG_AT_BUF_SIZE_EXP (7)
#endif

#define AT_BUF_SIZE   (1<<CONFIG_AT_BUF_SIZE_EXP)
```

Reply:
but this would make CONFIG_AT_BUF_SIZE_EXP default to 7 , right ?

Reply:
Done. Tested and Updated.